napari/plugins/_npe2.py

```diff
@@ -43,7 +43,7 @@ def read(
     """Try to return data for `path`, from reader plugins using a manifest."""
 
     # do nothing if `plugin` is not an npe2 reader
-    if plugin and plugin not in get_readers():
+    if plugin and not any(plugin.startswith(name) for name in get_readers()):
```

Comment:
This is super danger zone and is going to lead to the same issues as were fixed in #npe2/297. I suggest we explicitly perform a split on the `'.'` here and check the full plugin name:

```python
    # do nothing if `plugin` is not an npe2 reader
    if plugin:
        # user might have passed 'plugin.reader_contrib' as the command
        plugin = plugin.split('.')[0]
        if plugin not in get_readers():
            return None
```

I don't know if it should be our job to check down to the contribution level here - I've left that for `npe2` to check - note that without npe2#301, npe2 in this case may just use a different contribution... But I just don't really know if adding the check here is right...

Reply:
Makes sense. I implemented your suggestion, just using `partition('.')` because it seems safer in case of no delimiter and we only care about whats before the `.` anyways?
Tests pass with npe2 0.7.0 and my test case with napari builtin works.